godotopenxrmeta/src/main/cpp/include/openxr_fb_scene_extension_wrapper.h

```diff
@@ -0,0 +1,119 @@
+/**************************************************************************/
+/*  openxr_fb_scene_extension_wrapper.h                                   */
+/**************************************************************************/
+/*                       This file is part of:                            */
+/*                              GODOT XR                                  */
+/*                      https://godotengine.org                           */
+/**************************************************************************/
+/* Copyright (c) 2022-present Godot XR contributors (see CONTRIBUTORS.md) */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#pragma once
+
+#include <godot_cpp/classes/open_xr_extension_wrapper_extension.hpp>
+#include <openxr/openxr.h>
+#include <openxr/fb_scene.h>
+#include <godot_cpp/variant/utility_functions.hpp>
+
+#include "util.h"
+
+#include <optional>
+#include <map>
+
+using namespace godot;
+
+struct XrSceneObjectInternal {
+	XrUuidEXT uuid;
+	XrSpace space;
+	std::optional<std::string> label;
```

Comment:
Please use Godots implementations whenever possible to prevent needless conversions of data between internal data and Godot.

Reply:
This collection of OpenXR extensions use XrSpace and XrUuidEXT across multiple API calls to request different metadata and poses (for example, once you've queried for all available anchors, you can locate that anchor or ask about it's semantic labels / bounding volumes, all with uuid / space). The intent here is to store in the original OpenXR formats to facilitate later calls associated with the same anchor, and translate to a Godot type in some higher level "XrSceneManager" that gets used by gdscript (rather than using these wrappers directly)